pkg/server/node.go

```diff
@@ -451,25 +453,33 @@ func (n *Node) start(
 		}
 		return nil
 	}); err != nil {
-		return errors.Wrapf(err, "failed to read last up timestamp from stores")
+		return errors.Wrapf(err, "failed to read last up timestamp from stores"), nil
 	}
 	n.lastUp = mostRecentTimestamp.WallTime
 
 	// Set the stores map as the gossip persistent storage, so that
 	// gossip can bootstrap using the most recently persisted set of
 	// node addresses.
 	if err := n.storeCfg.Gossip.SetStorage(n.stores); err != nil {
-		return fmt.Errorf("failed to initialize the gossip interface: %s", err)
+		return fmt.Errorf("failed to initialize the gossip interface: %s", err), nil
 	}
 
-	// Bootstrap any uninitialized stores.
-	//
-	// TODO(tbg): address https://github.com/cockroachdb/cockroach/issues/39415.
-	// Should be easy enough. Writing the test is probably most of the work.
+	// Bootstrap any uninitialized stores and signal channel once complete.
+	waitForBootstrapNewStores = func() { }
 	if len(state.newEngines) > 0 {
-		if err := n.bootstrapStores(ctx, state.firstStoreID, state.newEngines, n.stopper); err != nil {
-			return err
-		}
+		// We need to bootstrap additional stores asynchronously because we can't rely on
+		// the KV layer to be available yet. See #39415.
+
+		// Create bootstrapNewStoresDone channel and update waitForBootstrapNewStores function to block on it.
```

Comment:
[nit] The comment here is far too identical to the code right below, so doesn't necessarily add much to it. Given our variable names (they're also documentation in some sense), I think it's fine to skip it entirely.